test/python/transpiler/test_high_level_synthesis.py

```diff
@@ -2309,6 +2309,22 @@ def test_qft_line_plugin_annotated_qft(self, qft_plugin_name):
         qct = hls_pass(qc)
         self.assertEqual(Operator(qc), Operator(qct))
 
+    @data("line", "full")
+    def test_skip_non_qft(self, qft_plugin_name):
+        """Test that synthesis plugins are not applied on gates that are called `qft`, yet
+        that are not of type `QFTGate`.
+        """
+        qc = QuantumCircuit(1)
+        qc2 = QuantumCircuit(1, name="qft")
+        qc2.s(0)
+        qc.append(qc2.to_instruction(), qc.qregs[0])
+        hls_config = HLSConfig(qft=[qft_plugin_name])
+        hls_pass = HighLevelSynthesis(hls_config=hls_config)
+        qct = hls_pass(qc)
+        # HighLevelSynthesis should replace the custom gate called "qft"
+        # by the user-provided definition.
+        self.assertEqual(Operator(qc2), Operator(qct))
```

Comment:
This test confused me at first, because I was expecting the test to be that `HighLevelSynthesis` had _not_ turned it into a QFT equivalent, but the trick is that `qc2` doesn't actually encode a QFT, so the test does the right thing.

I'm fine to merge like this, just commenting more because I got a bit confused and felt like broadcasting that fact.